### ISSUE TITLE
Large image files are not detected correctly

### ISSUE THREAD

**Issue by whocares0101** [joncampbell123/dosbox-x#1992]
While testing more CHD files I encountered a problem if the CHD file is large (>2GiB?)
this code fails on my windows build (stat returns -1) and will return 2 instead of 1 for the image file.
https://github.com/joncampbell123/dosbox-x/blob/master/src/dos/cdrom.cpp#L245

I got this working by using `_stat32i64` instead.
```
struct _stat32i64 file_stat;
if ((_stat32i64(path, &file_stat) == 0) && (file_stat.st_mode & S_IFREG)) return 1;
```

But then the CHD code failed to load these bigger files as well, I don't know if this is because of my build and needs some defines or is an error in libchdr.
https://github.com/rtissera/libchdr/issues/40

**Comment by Wengier** [comment]
@whocares0101 Good catch. You are certainly right that the ```CDROM_GetMountType``` function in cdrom.cpp did not work properly for large CD image files. But I think you can just change ```stat``` to ```pref_stat``` (which is defined in dosbox.h) in this case, which should work cross-platform. While the ```_stat32i64``` function does work in Windows, it does not appear to work in Linux/GCC for example.
